contracts/strategy/StrategyBaseV2.sol

```diff
@@ -155,56 +162,43 @@ abstract contract StrategyBaseV2 is IStrategyV2, ControllableV3 {
     emit ManualClaim(msg.sender);
   }
 
-  /// @dev In case of broken base amounts, operator can assign baseAmount = current balance
-  function resetBaseAmounts(address[] memory tokens_) external {
-    StrategyLib.onlyOperators(controller());
-    uint len = tokens_.length;
-    for (uint i; i < len; ++i) {
-      baseAmounts[tokens_[i]] = IERC20(tokens_[i]).balanceOf(address(this));
-    }
-  }
-
   // *************************************************************
   //                    DEPOSIT/WITHDRAW
   // *************************************************************
 
   /// @notice Stakes everything the strategy holds into the reward pool.
-  /// @param amount_ Amount transferred to the strategy balance just before calling this function
+  /// amount_ Amount transferred to the strategy balance just before calling this function
   /// @param updateTotalAssetsBeforeInvest_ Recalculate total assets amount before depositing.
   ///                                       It can be false if we know exactly, that the amount is already actual.
-  /// @return totalAssetsDelta The {strategy} can update its totalAssets amount internally before depositing {amount_}
-  ///                          Return [totalAssets-before-deposit - totalAssets-before-call-of-investAll]
+  /// @return strategyLoss Loss should be covered from Insurance
   function investAll(
-    uint amount_,
+    uint /*amount_*/,
     bool updateTotalAssetsBeforeInvest_
   ) external override returns (
-    int totalAssetsDelta
+    uint strategyLoss
   ) {
     StrategyLib.onlySplitter(splitter);
 
-    address _asset = asset;
-    uint balance = IERC20(_asset).balanceOf(address(this));
-
-    _increaseBaseAmount(_asset, amount_, balance);
+    uint balance = IERC20(asset).balanceOf(address(this));
 
     if (balance > 0) {
-      totalAssetsDelta = _depositToPool(balance, updateTotalAssetsBeforeInvest_);
+      strategyLoss = _depositToPool(balance, updateTotalAssetsBeforeInvest_);
     }
     emit InvestAll(balance);
 
-    return totalAssetsDelta;
+    return strategyLoss;
   }
 
   /// @dev Withdraws all underlying assets to the vault
-  /// @return Return [totalAssets-before-withdraw - totalAssets-before-call-of-withdrawAllToSplitter]
-  function withdrawAllToSplitter() external override returns (int) {
+  /// @return strategyLoss Loss should be covered from Insurance
+  function withdrawAllToSplitter() external override returns (uint strategyLoss) {
     address _splitter = splitter;
     address _asset = asset;
     StrategyLib.onlySplitter(_splitter);
 
     uint balance = IERC20(_asset).balanceOf(address(this));
 
-    (uint investedAssetsUSD, uint assetPrice, int totalAssetsDelta) = _withdrawAllFromPool();
+    (uint investedAssetsUSD, uint assetPrice, uint _strategyLoss) = _withdrawAllFromPool();
```

Comment:
investedAssetsUSD => expectedWithdrewUSD
It worth to rename investedAssetsUSD in StrategyLib.checkWithdrawImpact too